src/components/Tokens/TokenDetails/index.tsx

```diff
@@ -137,8 +170,8 @@ export default function TokenDetails({
                   <CurrencyLogo currency={token} size="32px" />
                   <L2NetworkLogo networkUrl={L2Icon} size="16px" />
                 </LogoContainer>
-                {token?.name ?? <Trans>Name not found</Trans>}
-                <TokenSymbol>{token?.symbol ?? <Trans>Symbol not found</Trans>}</TokenSymbol>
+                {token?.name ?? onChainToken?.name ?? <Trans>Name not found</Trans>}
```

Comment:
This is unnecessary - both should never be fetched, so token already includes this data.